VultisigApp/VultisigApp/Services/SwapService/SwapCoinsResolver.swift

```diff
@@ -38,6 +38,28 @@ struct SwapCoinsResolver {
     }
     
     static func resolveAllProviders(fromCoin: Coin, toCoin: Coin) -> [SwapProvider] {
-        return fromCoin.swapProviders.filter { toCoin.swapProviders.contains($0) }
+        var commonProviders = fromCoin.swapProviders.filter { toCoin.swapProviders.contains($0) }
+        
+        // If toCoin is thorchain stagenet, remove mainnet thorchain provider to avoid mixing networks
+        if toCoin.chain == .thorChainStagenet {
```

Comment:
Maybe we can unify the `ifs` here and have only 2?  
```suggestion
        if toCoin.chain == .thorChainStagenet  || fromCoin.chain == .thorChainStagenet {
```